packages/createrepo_c/createrepo_c.spec

```
@@ -1,53 +1,52 @@
-# explicitly define, as we build on top of an scl, not inside with scl_package
-%if 0%{?scl:1}
-%global scl_prefix %{scl}-
-%global python3_sitearch /opt/theforeman/tfm-pulpcore/root/usr/lib64/python3.8/site-packages/
-%global python3_version 3.8
-%global __os_install_post %python38_os_install_post
-%global __python_requires %python38_python_requires
-%global __python_provides %python38_python_provides
-%global __python3 /opt/rh/rh-python38/root/usr/bin/python3
-%endif
-
-# Our EL8 buildroots default to Python 3.8, but let's also build 3.6, just to be safe
-%if 0%{?rhel} == 8
-%bcond_without python36
-%else
-%bcond_with python36
-%endif
-
 %global libmodulemd_version 2.3.0
 
 %define __cmake_in_source_build 1
 
 %global bash_completion %{_datadir}/bash-completion/completions/*
 
-%if 0%{?rhel} && ( 0%{?rhel} <= 7 || 0%{?rhel} >= 9 )
+%if ( 0%{?rhel} && ( 0%{?rhel} <= 7 || 0%{?rhel} >= 9 ) ) || ( 0%{?fedora} && 0%{?fedora} >= 39 )
 %bcond_with drpm
 %else
 %bcond_without drpm
 %endif
 
-%if 0%{?rhel} || 0%{?fedora} < 29
+# Our EL8 buildroots default to Python 3.8, but let's also build 3.6, just to be safe
+%if 0%{?rhel} == 8
```

If you check if Python 3 isn't 3.6 it also works where buildroots don't have another Python module defined:
```suggestion
%if 0%{?rhel} == 8 && 0%{python3_pkgversion} != 3
```

---

Sorry, I should have tested this.
```suggestion
%if 0%{?rhel} == 8 && 0%{python3_pkgversion} != 03
```

---

Even with `&& 0%{python3_pkgversion} != 03` still fails in the verify version and release

---

What is interesting is that the line number is off. It's trying to parse the old file (for comparison), and failing.

---

Yes, this is weird

---

It's not weird: it's checking in `rpm/develop` and getting the version. Then it does the same in the PR. But that assumes you have working specs in `rpm/develop`.

For now I'd recommend a small PR that just makes sure `createrepo_c` works on plain EL8 and merge that, even if CI fails. Doesn't need to bump the release either and doesn't need to be built.

Then once it's merged, rebase this and I'd expect it to pass.

---

I just did a small PR, with the patch and it worked https://github.com/theforeman/pulpcore-packaging/pull/740